### PR TITLE
Add perchance-generator extension

### PR DESCRIPTION
**Pull request by Visual-Studio-Coder**
## Screencast
image
So here's what you do. Go to perchance.org, and click on "generators" and then get the endpoint of that generator URL. Now, you just input that into Perchance Generator for Raycast, and you're pretty much done.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — COMMENTED** [review]
Hi 👋 

How do I use this? I'm not sure what it does, could you create a screencast of the flow?

**Comment by Visual-Studio-Coder** [comment]
try inputting `plotgenertor` to this generator's argument

**Comment by pernielsentikaer** [comment]
Is there a reason why it takes to long time to generate? It take more than 15 seconds here

**Comment by Visual-Studio-Coder** [comment]
It's just how their api works unfortunately. It's faster on future requests

**Comment by pernielsentikaer** [comment]
Sounds like the content should be shown in a `<Detail>` view instead of a toast?

**Comment by Visual-Studio-Coder** [comment]
Well no. The reason is in a toast is because it was designed so that people can spam generations and only copy when they find something good. Sometimes it's also just for a quick laugh, so I wouldn't want a detail view

**Comment by pernielsentikaer** [comment]
You could use useFetch's revalidate inside the `<Detail>`, like I did with [Dad jokes](https://www.raycast.com/pernielsentikaer/dad-jokes)

**Comment by Visual-Studio-Coder** [comment]
Cmon. I really liked the toast messages style. Even the dice rolling extension uses it. Could I still keep the toast message layout or do I HAVE to change it to detail.

**Comment by pernielsentikaer** [comment]
There is way too much text for the toast, it's not making the best UI/UX, also, from a detail pane you can generate it again with just a hit on enter 🙂

Raycast 2023-09-15 at 11 29 01

**Comment by Visual-Studio-Coder** [comment]
Ok that makes sense. I'll work on fixing this

**Comment by Visual-Studio-Coder** [comment]
Ok. I just copied your interface and added some actions and an image.

File: extensions/perchance-generator/package.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "perchance-generator",
+  "title": "Perchance Generator",
+  "description": "Input the project name and receive the generated text.",
```
Comment (pernielsentikaer) [inline extensions/perchance-generator/package.json:5]: Project name?

File: extensions/perchance-generator/package.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    "random-text-generator",
+    "keyboard",
+    "words"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Generate Using Perchance",
+      "description": "Template with a plain detail view",
```
Comment (pernielsentikaer) [inline extensions/perchance-generator/package.json:28]: Should be changed

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Thanks for the update 🔥 

Right when entering the extension it shows the success toast, after waiting around 10 seconds the data is presented, could be good with isLoading and a toast saying loading instead

Is that possible?

File: extensions/perchance-generator/package.json (hunk 1 above)
Comment (Visual-Studio-Coder) [reply extensions/perchance-generator/package.json:5]: Yeah, I believe that's what they're called . . .

File: extensions/perchance-generator/package.json (hunk 2 above)
Comment (Visual-Studio-Coder) [reply extensions/perchance-generator/package.json:28]: Ok, I'll fix that.

File: extensions/perchance-generator/package.json (hunk 1 above)
Comment (Visual-Studio-Coder) [reply extensions/perchance-generator/package.json:5]: Hmmm. I did a little more digging and found that they also called them _generator names_ OR project names, so will change to first one

**Comment by Visual-Studio-Coder** [comment]
You spelled it wrong...

**Comment by Visual-Studio-Coder** [comment]
I think it might be an internet issue.

**Comment by Visual-Studio-Coder** [comment]
Wait sorry, its that "undefined" thing that I forgot to put in quotes. I fixed it

**Comment by pernielsentikaer** [comment]
I have made a few minor corrections. Could you please review to ensure that nothing has been broken? :)

**Comment by Visual-Studio-Coder** [comment]
Ok yeah, it works. Thanks for adding the corrections.

**Comment by Visual-Studio-Coder** [comment]
@pernielsentikaer

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**